Mage/src/main/java/mage/game/combat/Combat.java

```diff
@@ -644,6 +644,16 @@ protected boolean checkAttackRestrictions(Player player, Game game) {
                     }
                 }
             }
+            // Allow unconventional forced attacks (War's Toll) to record creatures as having been forced to attack.
```

Comment:
1. Workaround looks strange. I’ll look at it. Maybe all that checkAfter must be renamed or used another logic (it’s strange to call one time requirement effects and multiple times restriction effects)
2. Need tests.
3. Miss clear of creaturesForcedToAttack on removing from attack? Human player can select and unselect creature from attack. Make sure it’s clear restrictions on removing troll from attack.